packages/realm-react/src/UserProvider.tsx

```diff
@@ -35,30 +35,52 @@ type UserProviderProps = {
   children: React.ReactNode;
 };
 
+function userWasUpdated(userA: Realm.User | null, userB: Realm.User | null) {
+  if (!userA && !userB) {
+    return false;
+  } else if (userA && userB) {
+    return (
+      userA.id !== userB.id ||
+      userA.state !== userB.state ||
+      userA.accessToken !== userB.accessToken ||
+      userA.refreshToken !== userB.refreshToken
+    );
+  } else {
+    return true;
+  }
+}
+
 /**
  * React component providing a Realm user on the context for the sync hooks
  * to use. A `UserProvider` is required for an app to use the hooks.
  */
 export const UserProvider: React.FC<UserProviderProps> = ({ fallback: Fallback, children }) => {
   const app = useApp();
   const [user, setUser] = useState<Realm.User | null>(() => app.currentUser);
+  const [, forceUpdate] = useReducer((x) => x + 1, 0);
```

Comment:
More idiomatic to use a state setter here:
```
const [, forceUpdate] = useState(0);
const triggerUpdate = () => forceUpdate((prev) => prev + 1);
```

Reply:
Nice to see that you also notice that we'd like to keep the function reference the same 👍

Since the suggestion splits up the steps a bit more, is it purely from an idiomatic stand point you prefer that implementation? 🙂

Reply:
There's not much in this. I prefer the readability of not using useReducers here just to increase a counter, but it is an effective approach and stable function reference. Thanks for working on this issue 😄

Reply:
Note, this snippet was copied from the React documentation: https://legacy.reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdatea

Reply:
That being said, we could / should probably migrate this to use [useSyncExternalStore](https://react.dev/reference/react/useSyncExternalStore). What do you think @elle-j - perhaps for a future PR?

Reply:
Seems like that could be something to look into for sure 👍